pkg/agent/controller/traceflow/packetin.go

```diff
@@ -176,10 +180,22 @@ func (c *Controller) parsePacketIn(pktIn *ofctrl.PacketIn) (*opsv1alpha1.Tracefl
 				return nil, nil, err
 			}
 		}
-		if tunnelDstIP != "" && tunnelDstIP != c.nodeConfig.NodeIPAddr.IP.String() {
-			ob.TunnelDstIP = tunnelDstIP
+		var outputPort uint32
+		if match = getMatchRegField(matchers, uint32(openflow.PortCacheReg)); match != nil {
+			outputPort, err = getInfoInReg(match, nil)
+			if err != nil {
+				return nil, nil, err
+			}
+		}
+		if outputPort == config.DefaultTunOFPort || outputPort == config.HostGatewayOFPort {
```

Comment:
Question - if the destination IP of the TF packet is indeed the gw0 IP, what should be the expected behavior here? And the action should be forwarded or delivered?

Reply:
Currently we restrict destination must be Pod. If the destination is other IP, we cannot identify if the packet is properly delivered.

Reply:
In TraceflowSpec the destination could be an IP? And if it is the Pod IP, I think we still track "delivered"?

In theory we can handle the case destination IP is gw0 IP too. If you want not to support that in this PR, probably add a comment in InstallTraceflowFlows() to explain that.

Even when the IP is an external one, I think antrea-controller should handle it (as we know it is forwarded through gw0), and not to report timeout failure.

Reply:
We can handle this case by checking whether the dstIP is in Pod cache to identify if the action is Delivered or Forwarded, but I think we should submit another patch after this patch merged.